p2p/conn/evil_secret_connection_test.go

```diff
@@ -258,7 +259,6 @@ func TestMakeSecretConnection(t *testing.T) {
 		{"share bad ethimeral key", newEvilConn(true, true, false, false), func(err error) bool { return assert.Contains(t, err.Error(), "wrong wireType") }},
 		{"refuse to share auth signature", newEvilConn(true, false, false, false), func(err error) bool { return errors.Is(err, io.EOF) }},
 		{"share bad auth signature", newEvilConn(true, false, true, true), func(err error) bool { return errors.As(err, &ErrDecryptFrame{}) }},
-		{"all good", newEvilConn(true, false, true, false), func(err error) bool { return err == nil }},
```

Comment:
I have spent over 12 hours attempting to fix this test, and it is incredibly complicated to do so. (The hardest test change out of everything I've done in comment) I do not think its worth investing in fixing the test. 

And this is because this is mocking an invalid secret connection, and then a "final check" that we mocked the whole evil secret connection correctly. Which involves simulating all reads/writes to something internal.

After talking with @zmanian we both think this test is unneeded. This test is basically testing an evil secret connection peer, who sent bad signatures /deviated from protocol could not authenticate. We check the exact error at every step, so we are still matching every error case. So it seems fine to delete the test that the evil secret connection mocking, would work for the full connection. 

We test that good handshakes complete in many  ways (All E2E tests, other secret connection tests, many switch tests, etc) I have also tested this on mainnet on multiple occasions.

Reply:
cc @melekes if this sounds good to you. Your the one who wrote this set of tests!

Reply:
The produced error, which should be nil is:

```
SecretConnection: failed to decrypt the frame: chacha20poly1305: message authentication failed
```

Reply:
It is produced here: 

https://github.com/cometbft/cometbft/blob/033f7bbf4f5f8a97a78d82275d84f8deb9f8eecc/p2p/conn/secret_connection.go#L437

Reply:
Anyway, do you mind commenting this line out instead of removing it? And put a comment "fails with the introduction of changes PR #3419"?

Reply:
It's okay to comment out the test.